src/main/java/de/tum/in/www1/artemis/config/CacheConfiguration.java

```diff
@@ -102,6 +107,47 @@ public CacheManager cacheManager(HazelcastInstance hazelcastInstance) {
         return new HazelcastCacheManager(hazelcastInstance);
     }
 
+    /**
+     * This scheduled task regularly checks if all members of the Hazelcast cluster are connected to each other.
+     * This is one countermeasure to a split cluster.
+     */
+    @Scheduled(fixedRate = 2, initialDelay = 1, timeUnit = TimeUnit.MINUTES)
+    public void connectToAllMembers() {
+        if (registration == null) {
+            return;
+        }
+        String serviceId = registration.getServiceId();
+        var hazelcastInstance = Hazelcast.getHazelcastInstanceByName("Artemis");
```

Comment:
Is it a good idea to hardcode the Instance name here?

Reply:
This was already done in the `hazelcastInstance` method before, so I just reused it.  Though the correct config value was already present in this class, it just was not used here before 🙃 

Thanks for the hint I now improved this as well.